### PR TITLE
chore: update cosi module, rekres

### PR DESCRIPTION
**Pull request by Unix4ever**
Rekres updates the project to use go 1.21.

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,15 +1,15 @@
 module github.com/cosi-project/state-etcd
 
-go 1.20
+go 1.21.0
```
[the remark this answers is not in the document]
Comment (DmitriyMV) [reply go.mod:3]: Yes - before 1.21 the proper version was `y.xx`.

[the remark this answers is not in the document]
Comment (frezbo) [reply go.mod:3]: i meant do we really need to pin to minor, i had the same question on talos repo

[the remark this answers is not in the document]
Comment (Unix4ever) [reply go.mod:3]: Go manages that itself and starting from 1.21 it adds minor version to it

**Pull request merged**